Zwave documentation update (#14485)
home-assistant/home-assistant.github.io: source/_docs/z-wave/adding.markdown

description: "How to add and remove Z-Wave devices."
---

> **Recommendation before adding any devices:**

Reviewer: If you need to emphasize it like a header, make it a header.

```suggestion
## Recommendation before adding any devices
```

Author: will update in the next commit

> Z-Wave devices behave as a mesh and can store network relationship details on the device itself.  This means used devices or even brand new devices could already be enrolled in another network, for example a test network for a brand new device or a previous network for devices aren't new. This could cause headaches when you're attempting to add/enrol the device to your network. It is recommended that if possible, perform a factory reset AND device perform exclusion or disenroll for the device you're attempting to add to Home Assistant. Steps can found further below on this page under "Removing Devices".

Reviewer: ```suggestion
Z-Wave devices behave as a mesh and can store network relationship details on the device itself. This means used devices or even brand new devices could already be enrolled in another network, for example, a test network for a brand new device or a previous network for devices isn't new. This could cause headaches when you're attempting to add/enroll the device to your network. It is recommended that if possible, perform a factory reset AND device perform exclusion or disenroll for the device you're attempting to add to Home Assistant. Steps can found further below on this page under "Removing Devices".
```

Author: At a glance this looks like no changes were recommended? I'll resolve it for now but let me know if I should make changes, will be happy to.

Reviewer: there is a `,` comma missing after `, for example`, that is what this suggestion is adding.

Author: got it thanks! also updating a section that I said "Previous devices aren't new" that should be "previous devices that aren't new" pushing this commit

### Network Key

> An easy UNIX script to generate a random key:

Reviewer: Lets make it Linux, because technically we don't support the full Unix family.
Furthermore, the user might not be aware of the portability of this command to Linux.

Author: Adding to the next commit

> (remember you can run this in Home Assistant! just add the Terminal Add-on)

Reviewer: Changed to include the installation type names, removed the exclamation mark in the middle of the sentence and removed `just`, as is might not be `just` as easy for everybody).

```suggestion
(remember you can run this in Home Assistant OS or Supervised, you can use the Terminal Add-on)
```

Author: Thank you! updating in the next commit

To remove (exclude) a Z-Wave device from your system:

> 1. Go to the Z-Wave control panel in the Home Assistant frontend. (Home Assistant > Configuration > Integrations > Configure)

Reviewer: ```suggestion
1. Go to the Z-Wave control panel in the Home Assistant frontend. (Home Assistant > Configuration > Integrations > Z-Wave > Configure)
```

Author: updating

2. Click the **Remove Node** button in the *Z-Wave Network Management* card - this will place the controller in exclusion mode
3. Activate your device to be excluded by following the instructions provided with the device

> 4. The device should now be removed, but that won't show until you restart Home Assistant. Look for confirmation signal on the device if available, or confirm on the Home Assistant logs.

Reviewer: ```suggestion
4. The device should now be removed, but that won't show until you restart Home Assistant. Look for a confirmation signal on the device if available, or confirm on the Home Assistant logs.
```

Author: done

5. Run a *Heal Network* so all the other nodes learn about its removal

> 6. Make sure to Restart Home Assistant. Done through **Home Assistant > Configuration > Server Controls**

Reviewer: Why would we need to restart?

Author: Removing this line then if you believe it won't be required. I had just seen it referred as a common step for most configuration updates.

Reviewer: Configuration via the UI doesn't require restart.
